src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +41,27 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
+bool MetaClient::isMetadReady() {
+    if (sendHeartBeat_) {
+        auto ret = heartbeat().get();
+        if (!ret.ok()) {
+            LOG(ERROR) << "Heartbeat failed, status:" << ret.status();
+            ready_ = false;
+            return ready_;
+        }
+    }  // end if
     loadDataThreadFunc();
+    return ready_;
+}
+
+
+bool MetaClient::waitForMetadReady(int count, int retryIntervalSecs) {
+    int tryCount = count;
+    while (!isMetadReady() && ((count == -1) || (tryCount > 0))) {
+        --tryCount;
+        ::sleep(retryIntervalSecs);
+    }  // end while
+
```

Comment:
when count > 0, and ready_ is false after break the loop, we need start bgThread_ too?